### PR TITLE
Add group search

### PR DESCRIPTION
**Pull request by jarrydlee**

### CODE REVIEW

File: server/models/group.js [hunk 1]
```diff
@@ -119,6 +119,20 @@ GroupSchema.statics = {
     });
   },
 
+  search(name) {
```
Comment (kalebdavis) [inline server/models/group.js]: We call this method `search` but the only possible query parameter is name. I'd prefer it if we explicitly call the method `searchByName` or something similar so it's clear when it's used that this is only searching by name

**Comment by kwm4385** [comment]
Is this good to merge?

**Comment by jarrydlee** [comment]
@kalebdavis

**Comment by kwm4385** [comment]
LGTM at least

File: server/routes/group.js [hunk 2]
```diff
@@ -15,6 +15,9 @@ router.route('/:groupName')
   .put(validate(paramValidation.updateGroup), groupMiddleware.isGroupAdmin, groupCtrl.updateGroup)
   .delete(validate(paramValidation.byName), groupMiddleware.isGroupAdmin, groupCtrl.deleteGroup);
 
+router.route('/:groupName/searchByName')
```
Comment (kwm4385) [inline server/routes/group.js]: groupName doesn't make sense here

Comment (kwm4385) [inline server/routes/group.js]: something like `/group/search?name=something` makes more sense

Comment (jarrydlee) [inline server/routes/group.js]: Whoops

**Pull request merged**